### ISSUE TITLE
Broadcast diffrence shape tensor

### ISSUE THREAD

**Issue by lxl910915** [horovod/horovod#1509]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by tgaddair** [comment]
Hey @lxl910915, can you elaborate on your request?  It sounds like what you're requesting can be accomplished in three steps:

1. Broadcast the shape of the tensor to the other workers.
2. Allocate empty tensors of the correct shape on each of the receiving workers.
3. Broadcast the tensor to the other workers.

Is that the goal of this request?

**Comment by lxl910915** [comment]
> Hey @lxl910915, can you elaborate on your request? It sounds like what you're requesting can be accomplished in three steps:
> 
> 1. Broadcast the shape of the tensor to the other workers.
> 2. Allocate empty tensors of the correct shape on each of the receiving workers.
> 3. Broadcast the tensor to the other workers.
> 
> Is that the goal of this request?

Yes, this is our goal. But we don't known how to implement step 1 and step 2 in tensorflow. Can you give us an demo ? Thank you!

**Comment by lxl910915** [comment]
https://github.com/horovod/horovod/issues/248
